src/native/managed/cdacreader/src/Contracts/Thread.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+// TODO: [cdac] Add other counts / threads
+public record struct ThreadStoreData(
+    int ThreadCount,
+    TargetPointer FirstThread);
+
+internal class Thread : IContract
+{
+    public static Thread NotImplemented = new Thread();
+
+    public static IContract Create(Target target)
+    {
+        if (!target.TryGetContractVersion(nameof(Thread), out int version))
+            return NotImplemented;
+
```

Comment:
(nit, probably): add a `static virtual string ContractName {get; }` to `IContract` and move this code to `Registry.GetContract<T>` ?